float32[3] target_p
float32 duration 
---
string ack
